mumerge.py

```diff
@@ -488,7 +497,24 @@ def mu_dict_generator(tfit_filenames,
     
     return dict(tfit_dict)
 
-### USED IN LIKELIHOOD_CALCULATOR #############################################
+################################################################################################
+# This function will be used to filter out singletons and low coverage calls to increase overall
+# call quality. 
+def call_remover(mu_list):
+    '''
+    This function removes calls that only appear in one sample
+    (Later this will be the function that removes low quality/low
+    coverage calls as well)
+    '''
+    # Check whether there is more than 1 entry at that region (should we do this as >=1 region/replicate instead?)
+    if remove_singletons:
+        if len(mu_list) == 1: 
+            singletonfile.write("\n") # Write the singletons to a new output file (bad form not to define file until later...?)
+            singletonfile.write("\t".join([str(chromosome), 
```

Comment:
Regarding your reference to `singletonfile` ---while yes it works, it is not correct "pythonic" style. I tend to make all variables explicit, if possible, and make sure that they are previously defined. What you have works because it can't find `singletonfile` in the local name space, but when the function is called it's able to find the variable in the global namespace. Personally, I try to avoid relying on global namespace variables as much as possible. This makes it easier to read and avoids the potential for namespace collisions (also, I just have a hard time making sense of code, so it's simpler for me to understand). You do the same thing here with the `remove_singletons` flag. Personally, my preferred approach would be to define both these variables as explicit arguments in the `call_remover` function. replace the two instances of `call_remover(mu_list) with `call_remover(mu_list, remove_singletons, singletonfile)`.

Does that make sense?